### PR TITLE
OUT 438 API - Task labelling

### PR DESCRIPTION
**Pull request by aatbip**
### Tasks

- [API - Task labelling](https://linear.app/copilotplatforms/issue/OUT-438/api-task-labelling#update-5f96a5f2)

### Changes

- [x] Adds LabelMapping schema
- [x] Adds a new `label` column in the Task schema
- [x] Adds LabelMapping service that includes necessary code for generating labels for all the conditions

### CODE REVIEW

File: src/app/api/label-mapping/label-mapping.service.ts [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+import { CopilotAPI } from '@/utils/CopilotAPI'
+import { z } from 'zod'
+import { LabelMapping as PrismaLabelMapping } from '@prisma/client'
+
+export class LabelMapping extends BaseService {
+  /**
+   * This method returns the computed label
+   * @param assigneeId
+   * @param assigneeType
+   * @returns string
```
Comment (saugatmaharjan) [inline src/app/api/label-mapping/label-mapping.service.ts:12]: I see all the `return` statement inside `if` conditions so there a possibility that this function might return `void`

**Review by makeitraina — COMMENTED** [review]
One open question about the caching before. Is there a separate PR for that or did we just combine the tasks together?

**Comment by aatbip** [comment]
> One open question about the caching before. Is there a separate PR for that or did we just combine the tasks together?

@makeitraina Turning off caching related task was done in #91 which was check out from this branch. That PR was approved and merged into this branch that's why we are seeing the changes related to #91 in this branch. Out strategy is to create the children branch PR to merge against the parent branch if they are interrelated with each other.

File: src/app/page.tsx [hunk 2]
```diff
@@ -1,4 +1,5 @@
 export const dynamic = 'force-dynamic'
+export const fetchCache = 'force-no-store'
```
[the remark this answers is not in the document]
Comment (aatbip) [reply src/app/page.tsx:1]: I have now removed the unwanted caching parameters such as `cache:'no-store` and `revalidate:0`.

**Review by rrojan — APPROVED** [review]
LGTM :trophy:

**Pull request merged**